### PR TITLE
[RFC] Testlint

### PR DESCRIPTION
**Pull request by KillTheMule**
I find lints very usefull, but I noticed there are a lot of errors. I set out to correct them :)

This one makes one jump through some loops, since we want tabs in a multiline string. It's a large diff, but not a huge change: Substitute the first tab in each line by `\t`, and then adjust for that when passing the string to functions. There were some spaces at the end of lines, I figured they're not important and removed them.

@justinmk Would you be interested in merging this? If so, let me know, I'll lint all the other files, too, so testlint is clean again. If there's a reason not to, no problem, I'll just wait for your comment before putting work into this.

### CODE REVIEW

**Comment by bfredl** [comment]
What was wrong with the old formulation of the test?

**Comment by KillTheMule** [comment]
The testlint would go haywire about the mixture of tabs and spaces to indent. You can't see it well on github, just load that test file and `:set list`.

**Comment by bfredl** [comment]
Why not just keep the file excluded from `make testlint` then? Or if possible disable the indentation lint for this file specifically.

**Comment by KillTheMule** [comment]
Mainly because I did not think of it :D I'll investigate if I can do it, but imho it's not preferable to disable all lints for this file, so it's either disable the indentation lint specifically or this patch I'd say.

**Comment by bfredl** [comment]
If we need to make a file significantly harder to read to satisfy the lint the lint is not worth it.

**Comment by KillTheMule** [comment]
Ok, it's easily disabled inline, so I'm gonna use this to go through the tests.

**Comment by KillTheMule** [comment]
Appveyor failure is unrelated, something went wrong with 7z during the mingw32 build.

File: test/functional/legacy/003_cindent_spec.lua [hunk 1]
```diff
@@ -653,8 +654,8 @@ describe('cindent', function()
       }
 
       Constructor::Constructor(int a,
-                               int b 
-                              )  :
```
Comment (justinmk) [inline test/functional/legacy/003_cindent_spec.lua]: this text is a test fixture, it should not be linted

**Comment by justinmk** [comment]
other than `003_cindent_spec.lua` this looks good, and is appreciated :)

File: test/functional/legacy/003_cindent_spec.lua (hunk 1 above)
Comment (KillTheMule) [reply test/functional/legacy/003_cindent_spec.lua]: I'll remove this change, but could you kindly explain what a "test fixture" is? My engrish is lacking...

**Comment by KillTheMule** [comment]
CI Failures unrelated :)

File: test/functional/legacy/003_cindent_spec.lua (hunk 1 above)
Comment (justinmk) [reply test/functional/legacy/003_cindent_spec.lua]: "test fixture" is any resource needed to set up a test, e.g. sample text, a fake file, a fake plugin, etc.

**Comment by justinmk** [comment]
@KillTheMule I don't see any changes to `test/.luacheckrc`. Does this PR allow this line to be removed?

    files["functional/legacy"] = {ignore = { "611", "612", "613", "621" }}

**Comment by KillTheMule** [comment]
As far as I can see, that line did not work to begin with, otherwise most of the "problems" I fixed should not have occured. Maybe it failed to work at some point, and that's why the errors popped up?

**Comment by justinmk** [comment]
What do you mean? The lint build wouldn't pass if that line didn't work.

**Comment by KillTheMule** [comment]
You got me confused. `make testlint` throws errors on current master (which is why I set out to fix them). That means that this line in `test/.luacheckrc` did not work as expected. Or what am I missing?

**Comment by justinmk** [comment]
Not sure what's different. I just deleted the line in `test/.luacheckrc` and then `make testlint` fails as expected. Also tried with the latest https://github.com/mpeterv/luacheck , same results: `make testlint` succeeds on master unless I delete the line in `test/.luacheckrc`.

Anyways, if you delete the line in `test/.luacheckrc`, then post to this PR, we can see what CI thinks.

**Comment by KillTheMule** [comment]
Ok let's have a look. But locally, testlint fails even with the line present.

**Pull request merged**